### PR TITLE
fix kubetest for gs:// paths

### PR DESCRIPTION
**Pull request by BenTheElder**
`path.Dir` alone still gives you a `gs://` path (actually it munges `gs://` to `gs:/` too), whereas we want something we can `GET` in get-kube: this makes sure we form a base URL instead of a base path.

This also adds a unit test for this strategy.

### CODE REVIEW

File: kubetest/extract.go [hunk 1]
```diff
@@ -393,7 +393,8 @@ func (e extractStrategy) Extract(project, zone string) error {
 		}
 		return getKube(url, release)
 	case gcs:
-		return getKube(path.Dir(e.option), path.Base(e.option))
+		url := strings.Replace(path.Dir(e.option), "gs:/", "https://storage.googleapis.com/", 1)
```
Comment (ixdy) [inline kubetest/extract.go]: relying on `path.Dir` changing `gs://foo` to `gs:/foo` seems rather fragile (and also confusing - I didn't understand why this worked until I read the PR description).
maybe strip the leading `gs://`, call `path.Dir`, then prepend the https url?

Comment (BenTheElder) [reply kubetest/extract.go]: SGTM.

Comment (BenTheElder) [reply kubetest/extract.go]: Done.

File: kubetest/extract.go [hunk 2]
```diff
@@ -393,7 +393,10 @@ func (e extractStrategy) Extract(project, zone string) error {
 		}
 		return getKube(url, release)
 	case gcs:
-		return getKube(path.Dir(e.option), path.Base(e.option))
+		// strip gs://foo -> /foo
+		withoutGS := e.option[3:]
```
Comment (ixdy) [inline kubetest/extract.go]: nit: `strings.TrimPrefix` is a more self-documenting way of achieving same.

Comment (BenTheElder) [reply kubetest/extract.go]: makes sense, I figured that since the GCS case requires this prefix format it wasn't necessary.

**Pull request merged**